server/src/main/java/org/opensearch/action/search/SearchRequestContext.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.search;
+
+import org.apache.lucene.search.TotalHits;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class holds request-level context for search queries at the coordinator node
+ *
+ * @opensearch.internal
+ */
+public class SearchRequestContext {
+    private long absoluteStartNanos;
+    private final Map<String, Long> phaseTookMap;
+    private TotalHits totalHits;
+    private Map<String, Integer> shardStats;
+
+    public SearchRequestContext() {
+        this.phaseTookMap = new HashMap<>();
+        this.totalHits = null;
+    }
+
+    protected void updatePhaseTookMap(String phaseName, Long tookTime) {
+        this.phaseTookMap.put(phaseName, tookTime);
+    }
+
+    protected Map<String, Long> phaseTookMap() {
+        return phaseTookMap;
+    }
+
+    protected void setAbsoluteStartNanos(long absoluteStartNanos) {
```

Comment:
Considering that we are maintaining a requestContext now, we should initialize absoluteStartNanos in constructor itself. That way we every listener can use the same value. And then use it from a getter below.

As otherwise any listener implementation can modify this value which seems wrong.

Reply:
Done